mithril-test-lab/mithril-end-to-end/src/mithril/infrastructure.rs

```diff
@@ -96,9 +103,56 @@ impl MithrilInfrastructure {
                     .as_ref()
                     .to_string(),
             ),
+            current_era: config.mithril_era.clone(),
+            next_era: config.mithril_next_era.clone(),
+            era_reader_adapter: config.mithril_era_reader_adapter.clone(),
+            regenesis_on_era_switch: config.mithril_era_regenesis_on_switch,
         })
     }
 
+    async fn register_startup_era(
+        aggregator: &mut Aggregator,
+        config: &MithrilInfrastructureConfig,
+    ) -> StdResult<()> {
+        let era_epoch = Epoch(0);
+        if config.mithril_era_reader_adapter == "cardano-chain" {
+            assertions::register_era_marker(
+                aggregator,
+                &config.devnet,
+                &config.mithril_era,
+                &era_epoch,
+            )
+            .await?;
+            sleep(Duration::from_secs(5)).await;
```

Comment:
Is it possible to add a comment to explain the reason for the sleep?

Reply:
This was in the original code, but I have removed it as it is not used anymore :+1: